### PR TITLE
Get all details of the repositories of the user

### PR DESCRIPTION
**Pull request by A15h**
This PR solves issue #159 
Get all repo details of user

## Checklist
I have performed a self-review of my own code
I have commented my code to understand areas
Kindly mark this contribution under GSSOC' 23
I appreciate your contribution.

### CODE REVIEW

**Comment by A15h** [comment]
Hi @nikhil25803 
I have done the necessary changes. Please tell me if there is any issue that I can work on and if not please merge my PR.
Thank you

**Review by nikhil25803 — CHANGES_REQUESTED** [review]
Why are you adding it as a file, as it as a method on the `Users` class of the `github` module.

**Comment by A15h** [comment]
> Why are you adding it as a file, as it as a method on the `Users` class of the `github` module.

Can you please explain what exactly I have to change in my code? I am very confused and am not able to understand. If you could explain it would be very kind of you.

**Comment by A15h** [comment]
hi @nikhil25803 please reply

**Comment by nikhil25803** [comment]
Hey @A15h | Just check this file
https://github.com/Clueless-Community/scrape-up/blob/main/src/scrape_up/github/users.py

**Comment by nikhil25803** [comment]
Any update on this PR @A15h?

**Comment by A15h** [comment]
> Any update on this PR @A15h?

yes i am working on the changes i'll do a new pull request today.

**Comment by A15h** [comment]
hi @nikhil25803 
do you want me to write different functions for different tasks or just one main function get_all _repo_details?

File: issue#159/git_scraper.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+Original file is located at
+    https://colab.research.google.com/drive/1vVkTt6t8CcCS4tXkAsfDVjD9BkoYyqPC
+"""
+
+
+
+
+
+import requests
```
Comment (nikhil25803) [inline issue#159/git_scraper.py:14]: Again, why are you adding it as a separate file? Add it as a method of the `Users` class (of the GitHub module).

Check how the GitHub module is organized and how classes and methods work.

**Review by nikhil25803 — CHANGES_REQUESTED** [review]

**Comment by nikhil25803** [comment]
> hi @nikhil25803 do you want me to write different functions for different tasks or just one main function get_all _repo_details?

Add this as a method of the `Users` class. Got it?

**Comment by A15h** [comment]
> > hi @nikhil25803 do you want me to write different functions for different tasks or just one main function get_all _repo_details?
> 
> Add this as a method of the `Users` class. Got it?

yes I got it but to get all the repository details i will have to scrape the page of repositories and for that i will have to call another function to scrape the repositories page and then a function which will use this function to get all the details.

**Comment by nikhil25803** [comment]
In the `Users` class only, you will find all the methods as per the requirements, try using them only.

You can use an asynchronous approach to make the faster.

**Pull request closed without merging**